src/data/repositories/UserD2ApiRepository.ts

```diff
@@ -176,6 +179,50 @@ export class UserD2ApiRepository implements UserRepository {
         return $requests.map(_.flatten);
     }
 
+    private addGroupsToUsers(users: User[], d2UsersWithGroups: D2UserGroupByKey): User[] {
+        return users.map((user): User => {
+            const userGroups = d2UsersWithGroups[user.id] || [];
+            return { ...user, userGroups: userGroups };
+        });
+    }
+
+    private getUsersGroups(usersIds: Id[]): FutureData<D2UserGroupByKey> {
+        const $requests = chunkRequest(usersIds, usersChunksIds => {
+            return apiToFuture(
+                this.api.models.userGroups.get({
+                    filter: { "users.id": { in: usersChunksIds } },
+                    fields: { id: true, displayName: true, users: true },
+                    paging: false,
+                })
+            ).map(response => {
+                return response.objects.map(d2Group => ({
+                    id: d2Group.id,
+                    name: d2Group.displayName,
+                    users: d2Group.users,
+                }));
+            });
+        });
+
+        return $requests.map(d2UsersGroups => {
+            const userGroups = _(d2UsersGroups)
+                .flatMap(group => group.users.map(user => ({ userId: user.id, group })))
+                .value();
+
+            const groupedByUser = _(userGroups)
+                .groupBy(ug => ug.userId)
+                .mapValues(groups =>
+                    groups.map(g => {
+                        /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
```

Comment:
To avoid "eslint-disable", 2 ways:

- `const { users: _users, ...groupWithoutUsers } = g.group;`
- `const groupWithoutUsers = _.omit(g.group, ["users"]); `